ShopAppTests/App/Account/Providers/AccountTableProviderSpec.swift

```diff
@@ -0,0 +1,129 @@
+            }
+            
+            it("should return correct footer class type") {
+                let footer = tableProvider.tableView(tableView, viewForFooterInSection: 0)
+                
+                expect(footer).to(beAnInstanceOf(UIView.self))
+            }
+        }
+        
+        describe("when policies did set") {
+            var policies: [Policy]!
+            
+            beforeEach {
+                policies = [Policy()]
+                tableProvider.policies = policies
+            }
+            
+            it("should return correct rows count") {
+                let rowsCount = tableProvider.tableView(tableView, numberOfRowsInSection: 0)
+                
+                expect(rowsCount) == policies.count
+            }
+
+            it("should return correct cell class") {
+                let indexPath = IndexPath(row: 0, section: 0)
+                let cell = tableProvider.tableView(tableView, cellForRowAt: indexPath)
+                
+                expect(cell).to(beAnInstanceOf(AccountTableViewCell.self))
+            }
+        }
+        
+        describe("when customer did set") {
+            beforeEach {
+                tableProvider.customer = Customer()
+            }
+            
+            it("should return correct header height") {
+                let headerHeight = tableProvider.tableView(tableView, heightForHeaderInSection: 0)
+                
+                expect(headerHeight) == kAccountLoggedHeaderViewHeight
+            }
+            
+            it("should return correct footer height") {
+                let footerHeight = tableProvider.tableView(tableView, heightForFooterInSection: 0)
+                
+                expect(footerHeight) == kAccountFooterViewHeight
+            }
+            
+            it("should return correct header class type") {
+                let header = tableProvider.tableView(tableView, viewForHeaderInSection: 0)
+                
+                expect(header).to(beAnInstanceOf(AccountLoggedHeaderView.self))
+            }
+            
+            it("should return correct footer class type") {
+                let footer = tableProvider.tableView(tableView, viewForFooterInSection: 0)
+                
+                expect(footer).to(beAnInstanceOf(AccountFooterView.self))
+            }
+        }
+        
+        describe("when cell selected") {
+            it("needs to show policy") {
+                let policies = [Policy()]
+                tableProvider.policies = policies
+                
+                let delegateMock = AccountTableProviderDelegateMock()
+                tableProvider.delegate = delegateMock
+                
+                let indexPath = IndexPath(row: 0, section: 0)
+                tableProvider.tableView(tableView, didSelectRowAt: indexPath)
+                
+                expect(delegateMock.provider) === tableProvider
+                expect(delegateMock.policy).toNot(beNil())
```

Comment:
What about create policy property, pass it to table provider and then check not only if selected policy not nil, also / instead check that selected policy the same with policy property we passed?